packages/medusa/src/loaders/custom-api.ts

```diff
@@ -0,0 +1,56 @@
+import { Express, NextFunction, Request, Response, Router } from "express"
+import { ConfigModule } from "../types/global"
+
+enum AllowedMethods {
+  GET = "GET",
+  POST = "POST",
+  DELETE = "DELETE",
+}
+
+type Handler = (
+  req: Request,
+  res: Response,
+  next: NextFunction
+) => Promise<void>
+
+export type MedusaRoute = {
+  name: string
+  method: AllowedMethods
+  path: string
+  handler: Handler | Handler[]
+  routes: MedusaRoute[]
+  options: {}
+}
+
+export default ({
+  app,
+  configModule,
+}: {
+  app: Express
+  configModule: ConfigModule
+}) => {
+  configModule.routes.map(async (customRoute) => {
+    let { handler, path, method, routes, name } = customRoute
+
+    if (!method || !path || !handler || !name) {
+      console.warn(`Invalid config for custom route: ${name}`)
```

Comment:
This should be a logger in order to output when starting the server. 

We should also add a more nuanced validation here since this will raise a warning with valid configurations such as a route with `routes` defined but no method, however method doesn't make a lot of sense when defining an array of subroutes. 

Specifically if `routes` are defined we should only validate `path` and `routes` exist.

if `routes` are not defined we should validate `method`, `path`, and `handler`.